webapp/js/components/Utils.js

```diff
@@ -54,3 +72,82 @@ var getQueryString = function ( field, url ) {
     var string = reg.exec(href);
     return string ? string[1] : null;
 };
+
+export function getAuthToken(callback) {
+
+    if (localStorage.getItem('token')) {
+        var t = JSON.parse(localStorage.getItem('token'))
+        var utcTimestamp = Math.floor((new Date()).getTime() / 1000)
+        if (t.exp > utcTimestamp) {
+            // Use the token from local storage
+            callback(t)
+            return
+        }
+    }
+
+    // Get a fresh token and store it in local storage
+    Ajax.getAuthToken().then((resp) => {
+        var jwt = resp.headers.authorization
+
+        if (!jwt) {
+            callback({})
+            return
+        }
+        var token = jwtDecode(jwt)
+
+        if (!token) {
+            callback({})
+            return
+        }
+
+        localStorage.setItem('token', JSON.stringify(token))
+        callback(token)
+    })
+
+}
+
+export function authToken() {
+    if (localStorage.getItem('token')) {
+        var t = JSON.parse(localStorage.getItem('token'))
+        return t
+    } else {
+        return {}
+    }
+}
+
+export function isLoggedIn(token) {
+    return isUserStandard(token)
+}
+
+export function isUserStandard(token) {
+    if (!token) return false
+    if (!token.role) return false
+
+    if (token.role >= Role.Standard) {
+        return true
+    } else {
+        return false
+    }
+}
+
+export function isUserAdmin(token) {
+    if (!token) return false
+    if (!token.role) return false
+
+    if (token.role >= Role.Admin) {
+        return true
+    } else {
+        return false
+    }
+}
+
+export function isUserSuperuser(token) {
```

Comment:
why 3 methods for the same thing? If that's just for clarity, why not better:
export function isUserAdmin(token) {
   return isUser(admin, token)
}
export function isUserStandard(token) {
   return isUser(stadandard, token)
}
export function isUserSuperuser(token) {
   return isUser(superuser, token)
}
export function isUser(role, token) {
    if (!token) return false
    if (!token.role) return false

    if (token.role >= role) {
        return true
    } else {
        return false
    }
}